### PR TITLE
Add SecretsManager, issue #105

### PR DESCRIPTION
**Pull request by geoffdutton**
This is an attempt at #105.

### CODE REVIEW

File: packages/laconia-config/src/SecretsManagerConfigConverter.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  async _getParameterMap(secretIds) {
+    const datas = await Promise.all(
+      secretIds.map(secretId =>
+        this.secretsManager
+          .getSecretValue({
+            SecretId: secretId
+          })
+          .promise()
+          .catch(err => {
+            if (err.code === "ResourceNotFoundException") {
```
Comment (ceilfors) [inline packages/laconia-config/src/SecretsManagerConfigConverter.js]: What's the reason of hiding this exception? I feel that if I'm the developer, I would like to know if I have misconfigured / mistyped the secret id.

File: packages/laconia-config/src/index.js [hunk 2]
```diff
@@ -2,10 +2,12 @@ const EnvVarConfigFactory = require("./EnvVarConfigFactory");
 const BooleanConfigConverter = require("./BooleanConfigConverter");
 const SsmConfigConverter = require("./SsmConfigConverter");
 const S3ConfigConverter = require("./S3ConfigConverter");
+const SecretsManagerConfigConverter = require("./SecretsManagerConfigConverter");
 
-exports.envVarInstances = () => ({ env, $ssm, $s3 }) =>
+exports.envVarInstances = () => ({ env, $ssm, $s3, $secretsManager }) =>
```
Comment (ceilfors) [inline packages/laconia-config/src/index.js]: `packages/laconia-core/src/laconia.js` would need to be updated to instantiate and register `$secretsManager`

File: packages/laconia-config/src/SecretsManagerConfigConverter.js (hunk 1 above)
Comment (geoffdutton) [reply packages/laconia-config/src/SecretsManagerConfigConverter.js]: I can't remember what I was thinking... I agree with you and am I going to remove it.

File: packages/laconia-config/src/index.js (hunk 2 above)
Comment (geoffdutton) [reply packages/laconia-config/src/index.js]: Is this what you mean, or am I missing a spot? https://github.com/geoffdutton/laconia/blob/issue-105-secrets-access/packages/laconia-core/src/laconia.js#L18

Comment (ceilfors) [reply packages/laconia-config/src/index.js]: My bad! I totally missed this :)

File: packages/laconia-config/src/SecretsManagerConfigConverter.js (hunk 1 above)
Comment (ceilfors) [reply packages/laconia-config/src/SecretsManagerConfigConverter.js]: I guess if we're not sure, I would suggest for it to fail by default even in the case of multiple secrets. Perhaps we can introduce a way to suppress it in the future if someone came along with a valid use case?

Comment (geoffdutton) [reply packages/laconia-config/src/SecretsManagerConfigConverter.js]: Yeah I agree with you. I don't have any use cases in mind. It's easy enough to address it if someone comes up with a use case.

Comment (ceilfors) [reply packages/laconia-config/src/SecretsManagerConfigConverter.js]: Great, I'll release this first thing in the morning tomorrow!

**Comment by ceilfors** [comment]
Released under 1.3.0. Thanks again for your contribution!

**Comment by ceilfors** [comment]
I've raised #166 and #167 as follow-ups. Feel free to contribute if you're interested!

**Pull request merged**